tests/pytests/test_vecsim.py

```diff
@@ -1383,23 +1383,129 @@ def test_timeout_reached():
         res = conn.execute_command('FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_vec,
                                    'PARAMS', 4, 'K', n_vec, 'vec_param', query_vec.tobytes(),
                                    'TIMEOUT', 1)
-        env.assertEqual(res[0], timeout_expected)
+        env.assertEqual(res[0], timeout_expected, message=algo)
 
         # HYBRID MODES
         for mode in hybrid_modes:
             res = conn.execute_command('FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]', 'NOCONTENT', 'LIMIT', 0, n_vec,
                                        'PARAMS', 6, 'K', n_vec, 'vec_param', query_vec.tobytes(), 'hp', mode,
                                        'TIMEOUT', 0)
-            env.assertEqual(res[0], n_vec)
+            env.assertEqual(res[0], n_vec, message=f'{algo}, {mode}')
 
             res = conn.execute_command('FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]', 'NOCONTENT', 'LIMIT', 0, n_vec,
                                        'PARAMS', 6, 'K', n_vec, 'vec_param', query_vec.tobytes(), 'hp', mode,
                                        'TIMEOUT', 1)
-            env.assertEqual(res[0], timeout_expected)
+            env.assertEqual(res[0], timeout_expected, message=f'{algo}, {mode}')
 
         conn.flushall()
+
+def test_create_multi_value_json():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+    prefix = '_' if env.isCluster() else ''
+    dim = 4
+    algos = ['FLAT', 'HNSW']
+    multi_paths = ['$..vec', '$.vecs[*]', '$.*.vec']
+
+    path = 'not a valid path'
+    env.expect('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA', path, 'AS', 'vec', 'VECTOR', 'FLAT',
+               '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2',).error().equal(
+                f"Invalid JSONPath '{path}' in attribute 'vec' in index 'idx'")
+
+    for algo in algos:
+        for path in multi_paths:
+            conn.flushall()
+            env.expect('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA', path, 'AS', 'vec', 'VECTOR', algo,
+                       '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2',).ok()
+            env.assertEqual(to_dict(env.cmd(prefix+"FT.DEBUG", "VECSIM_INFO", "idx", "vec"))['IS_MULTI_VALUE'], 1, message=f'{algo}, {path}')
+
+def test_index_multi_value_json():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+    dim = 4
+    n = 100
+    per_doc = 5
+    info_type = int if env.isCluster() else str
+
+    env.expect('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA',
+               '$.vecs[*]', 'AS', 'hnsw', 'VECTOR', 'HNSW', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2',
+               '$.vecs[*]', 'AS', 'flat', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2').ok()
+
+    for i in range(n):
+        conn.json().set(i, '.', {'vecs': [[0.46 for _ in range(dim)] for _ in range(per_doc)]})
```

Comment:
```suggestion
        conn.json().set(i, '.', {'vecs': [[0.46]*dim]* per_doc]})
```
you can use this syntatic sugar, sugar